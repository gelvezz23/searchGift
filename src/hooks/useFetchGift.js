/** @format */

import { useEffect, useState } from 'react';
import { getGift } from './../helpers/getGift';

export const useFetchGift = (category) => {
	const [state, setstate] = useState({
		data: [],
		loading: true,
	});

	useEffect(() => {
		getGift(category).then((imags) =>
			setstate({
				data: imags,
				loading: false,
			})
		);
	}, [category]);
	return state;
};

export default useFetchGift;
