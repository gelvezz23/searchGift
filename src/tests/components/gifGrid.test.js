/** @format */

import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';
import useFetchGift from '../../hooks/useFetchGift';
import GifGrid from './../../components/gifGrid';

// simular una llamada a un archivo
jest.mock('../../hooks/useFetchGift');

describe('Pruebas en componente <gifGrid/>', () => {
	const category = 'hunter x hunter';
	let wrapper;

	test('El componente de visualiza correctamente', () => {
		useFetchGift.mockReturnValue({
			data: [],
			loading: true,
		});
		wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar items cuando se le cargan imagenes con el customHook', () => {
		const gifs = [
			{
				id: 'ABC',
				image: 'https://localhost/algo.jpg',
				title: 'algo que no se',
			},
		];
		useFetchGift.mockReturnValue({
			data: gifs,
			loading: false,
		});
		wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').exists()).toBe(false);
		// cuantos componentes de imagenes hay renderizandoce
		expect(wrapper.find('GifItem').length).toBe(gifs.length);
	});
});
