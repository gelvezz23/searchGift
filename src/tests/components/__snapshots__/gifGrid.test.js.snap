// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <gifGrid/> Debe de mostrar items cuando se le cargan imagenes con el customHook 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    hunter x hunter
  </h3>
  <div
    className="card-grid animate__animated animate__fadeInDown"
  >
    <GifItem
      id="ABC"
      image="https://localhost/algo.jpg"
      key="ABC"
      title="algo que no se"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en componente <gifGrid/> El componente de visualiza correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    hunter x hunter
  </h3>
  <p>
    Loading ...
  </p>
  <div
    className="card-grid animate__animated animate__fadeInDown"
  />
</Fragment>
`;
